TestModels/SQSExtended/src/SQSExtendedImpl.dfy

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+// Do not modify this file. This file is machine generated, and any changes to it will be overwritten.
+include "../Model/PolymorphTutorialSqsextendedTypes.dfy"
+module AmazonSQSExtendedImpl refines AbstractPolymorphTutorialSqsextendedOperations {
+  datatype Config = Config(
+    nameonly sqsClient : ComAmazonawsSqsTypes.ISQSClient
+  )
+  type InternalConfig = Config
+  predicate ValidInternalConfig?(config: InternalConfig)
+  {
+    config.sqsClient.ValidState()
+  }
+  function ModifiesInternalConfig(config: InternalConfig): set<object>
+  {
+    config.sqsClient.Modifies
+  }
+  predicate HandleMessagesEnsuresPublicly(input: HandleMessagesInput , output: Result<(), Error>)
+  {true}
+
+
+
+  method HandleMessages ( config: InternalConfig , input: HandleMessagesInput )
+    returns (output: Result<(), Error>)
+
+  {
+    var queueUrl := input.receiveRequest.QueueUrl;
+
+    var maybeMessagesResult := config.sqsClient.ReceiveMessage(input.receiveRequest);
+    var messagesResult :- maybeMessagesResult.MapFailure(e => ComAmazonawsSqs(e));
+    // TODO: Handle error better
+    expect messagesResult.Messages.Some?;
+    var messages :- expect messagesResult.Messages;
+
+    for i := 0 to |messages| {
+      var message := messages[i];
+      // TODO: Propogate error instead of terminating
+      var receiptHandle :- expect message.ReceiptHandle;
```

Comment:
I think you should just be able to `MapFailure` here. The smithy-dafny error type should have a joining error that should just work.

non-blocking point :)

Reply:
Yup agreed. I'll address that when I follow up with adding more target languages.